### DISCUSSION TITLE
What's the correct way to override the CSS modules default "localIdentName"?

### DISCUSSION THREAD

**Discussion by let-lc** [vercel/next.js#20876] [Help]
Hi all,

My Next.js project is configured with Typescript and Tailwind CSS. I am adding all the Tailwind classes in a separated SCSS file and imported as a CSS modules just for separation of concerns and make the HTML code of the component cleaner.

What I want to do is to generate "hash only" CSS module classes in the production build. I basically looked at every issues and discussions of this repository that's related to this topic and also stackoverflow, tried everything but I couldn't find a solution for this problem. Sorry I am very new the Next.js and Webpack.

The closest solution that I found is from #14156, provided by mattcarlotta. This solution indeed can alter the default `localIdentName` config of the `css-loader`, but it is causing another bug for me.

The bug is all the classes are missing from the HTML in the browser, or shown as "undefined".

For example, in the source code:
```javascript
import styles from "./HelloWorld.module.scss";

export default HelloWold = () => {
    return <p className={styles.hello}>Hello World</p>
};
```

When it is rendered in the browser:
```html
<div id="__next">
    <p>Hello World</p>
<div>
```

The class name is missing from the `<p>` tag, I checked the css from the `<style>` tag in the browser, the solution by mattcarlotta is working, all the classes are in hash now. But the CSS is not applying because the class names are missing from the HTML.

I added to `console.log(styles)` to the component code to see what happened. In the browser console, I could see the styles object is printed as the following for example:

```javascript
{
    ...
    hello: "_1TBiY",
    ...
}
```

But in the system terminal, it prints the following:
```javascript
[
    [
        './components/HelloWorld.module.scss',
        '<all the css code from the HelloWorld.module.scss>',
        '',
        {
            version: 3,
            sources: [Arrays],
            names: [],
            mappings: '<Here's a long string with random hash>',
            sourcesContent: [Array],
            sourceRoot: ''
        }
    ],
    toString: [Function: toString],
    i: [Function],
    locals: {
        ...
        hello: "_1TBiY",
        ...
    }
]
```

In the browser console, it also print the following warning for other components:
```Warning: Prop `className` did not match. Server: "null" Client: "_3BnGy"```

Appreciation to anyone knows a solution for this issue, or a better way to override `localIdentName` without causing any issues.

**Comment by facuparedes** [comment]
Hey! Did you find a fix?

**Reply by let-lc** [reply]
Not actually a fix but something like a workaround solution. Overriding Next.js `getLocalIdent` function, it works for me. You can take a look on this discussion #15818, solution by DmitrySkripunov.
